modules/extensions-spring-support/src/main/java/org/mule/module/extension/internal/config/XmlExtensionParserUtils.java

```diff
@@ -0,0 +1,491 @@
+
+        if (value == null)
+        {
+            return null;
+        }
+
+        if (value instanceof String)
+        {
+            SimpleDateFormat format = new SimpleDateFormat(parseFormat);
+            try
+            {
+                return format.parse((String) value);
+            }
+            catch (ParseException e)
+            {
+                throw new IllegalArgumentException(String.format("Could not transform value '%s' into a Date using pattern %s", value, parseFormat));
+            }
+        }
+
+        if (value instanceof Date)
+        {
+            return (Date) value;
+        }
+
+        throw new IllegalArgumentException(
+                String.format("Could not transform value of type '%s' to Date", value != null ? value.getClass().getName() : "null"));
+    }
+
+    private static ValueResolver parseCalendar(ElementDescriptor element, String attributeName, DataType dataType, Object defaultValue)
+    {
+        Object value = getAttributeValue(element, attributeName, defaultValue);
+        if (isExpression(value, parser))
+        {
+            return new EvaluateAndTransformValueResolver((String) value, dataType);
+        }
+
+        Date date = doParseDate(element, attributeName, CALENDAR_FORMAT, defaultValue);
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+
+        return new StaticValueResolver(calendar);
+    }
+
+    private static ValueResolver parseDate(ElementDescriptor element, String attributeName, DataType dataType, Object defaultValue)
+    {
+        Object value = getAttributeValue(element, attributeName, defaultValue);
+        if (isExpression(value, parser))
+        {
+            return new EvaluateAndTransformValueResolver((String) value, dataType);
+        }
+        else
+        {
+            return new StaticValueResolver(doParseDate(element, attributeName, DATE_FORMAT, defaultValue));
+        }
+    }
+
+    static void setNoRecurseOnDefinition(BeanDefinition definition)
+    {
+        definition.setAttribute(MuleHierarchicalBeanDefinitionParserDelegate.MULE_NO_RECURSE, Boolean.TRUE);
+    }
+
+    static ValueResolver parseParameter(ElementDescriptor element, Parameter parameter)
+    {
+        return parseElement(element, parameter.getName(), parameter.getType(), parameter.getDefaultValue());
+    }
+
+    static ValueResolver parseElement(final ElementDescriptor element,
+                                      final String fieldName,
+                                      final DataType dataType,
+                                      final Object defaultValue)
+    {
+        final String hyphenizedFieldName = hyphenize(fieldName);
+        final String singularName = singularize(hyphenizedFieldName);
+        final ValueHolder<ValueResolver> resolverReference = new ValueHolder<>();
+
+        DataQualifierVisitor visitor = new BaseDataQualifierVisitor()
```

Comment:
same answer as before. There's an issue for that but I think that not everything is unit test material